src/main/java/uk/gov/pay/directdebit/mandate/services/MandateCommandService.java

```diff
@@ -0,0 +1,9 @@
+package uk.gov.pay.directdebit.mandate.services;
+
+import uk.gov.pay.directdebit.gatewayaccounts.model.GatewayAccount;
+import uk.gov.pay.directdebit.mandate.api.ConfirmMandateRequest;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+
+public interface MandateCommandService {
+    void confirm(GatewayAccount gatewayAccount, Mandate mandate, ConfirmMandateRequest confirmDetailsRequest);
```

Comment:
could add `create` and potentially `collect` to this interface

Reply:
We had it like that, but then I've realised (see description of the PR) that it adds quite a bit of complexity, because create and collect are used only for specific kind of mandates.
- collect is used only for on demand
- create is used only for one-off in the TransactionResource
- create is used only for on demand in the create agreement endpoint 

So, instead of getting the factory and switching on mandate type, I've used the specific mandate service right there, and that makes adding create and collect to the common interface pointless (for now)

Reply:
fair enough. I can def see your point with `collect`, but `create` seems naturally part of the interface, even if doesn't need to be from the point of view of how you've used it. But now I'm just imposing my own prejudices

Reply:
I think you're right, as in collect definitely feels wrong to have it in the interface (because it's not common), whereas create could be there even if it's not really used that way. I'll add it.

Reply:
sure